docs/source/background/overview.rst

```diff
@@ -0,0 +1,124 @@
+..
+.. Unless required by applicable law or agreed to in writing, software
+.. distributed under the License is distributed on an "AS IS" BASIS,
+.. WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+.. See the License for the specific language governing permissions and
+.. limitations under the License.
+
+.. _pp_overview:
+
+###################
+PluginPlay Overview
+###################
+
+The background section of the PluginPlay documentation is extensive. This is
+because PluginPlay is targeted at scientific research software and thus it
+is important to understand the scientific merits of PluginPlay and how it
+relates to other software. That said most users can safely ignore most of the
+content in the background section, this overview page is designed to give you
+the essentials in about a 5 minute read.
+
+*******************
+What is PluginPlay?
+*******************
+
+There is a need for modular scientific software. Using PluginPlay, developers
+write algorithms inside self-contained :ref:`module` objects. Software packages
+are then created by "wiring" :ref:`module` objects together. Additional
+:ref:`module` objects are developed as new methods and properties are
+discovered, the ecosystem of :ref:`module` objects grows, and the cycle
+repeats. PluginPlay facilitates writing sustainable modular scientific software
+packages by providing a framework that is focused on enabling dynamic:
+
+#. loading of :ref:`module` objects,
+#. interactions with :ref:`module` objects, and
+#. wiring of :ref:`module` objects.
+
+The dynamic nature of the resulting package makes it easier to extend the
+package to new use cases and new hardware. PluginPlay includes a number of
+other features designed to make development as easy as possible, *e.g.*, tools
+for:
+
+#. debugging/profiling :ref:`module` objects,
+#. recording provenance of the resulting program,
+#. creating :ref:`module` objects from existing code, and
+#. leveraging :ref:`module` objects in other code.
+
+
+**************************
+Who should use PluginPlay?
+**************************
+
+PluginPlay is meant as a framework on which to build a software package.
+PluginPlay is thus really targeted at developers who fall into two groups. The
+first group of developers actively write and maintain modular software
+package(s). The second group of developers write modules which can be consumed
+as plugins to the aforementioned modular software packages. Of course, these
+groups of developers need not be disjoint and developers may find themselves
+in both groups at one time or another.
+
+While PluginPlay can be used as a workflow tool, its domain-agnostic nature
+can make standard workflow tasks cumbersome. So while end-users of a software
+package can (and should be able to) interact directly with PluginPlay, in many
+cases developers can serve their community better by wrapping the PluginPlay
+calls needed for domain-specific tasks in domain-specific :ref:`api`.
```

Comment:
Sorry if I missed this, but does that link point to our domain specific api in NWChemEx as an example?

Reply:
No it points to the definition of API.